Dependencies/AndroidExtensions/Source/JavaWrappers.cpp

```diff
@@ -48,16 +48,56 @@ namespace java::lang
 
     Class::Class(const char* className)
         : m_env{GetEnvForCurrentThread()}
-        , m_class{m_env->FindClass(className)}
+        , m_class{static_cast<jclass>(m_env->NewGlobalRef(m_env->FindClass(className)))}
     {
     }
 
     Class::Class(const jclass classObj)
         : m_env{GetEnvForCurrentThread()}
-        , m_class{classObj}
+        , m_class{static_cast<jclass>(m_env->NewGlobalRef(classObj))}
     {
     }
 
+    Class::~Class()
+    {
+        JClass(nullptr);
+    }
+
+    Class::Class(const Class& other)
+        : Class(other.m_class)
+    {
+    }
+
+    Class& Class::operator=(const Class& other)
+    {
+        if (this != &other)
+        {
+            JClass(nullptr);
```

Comment:
Maybe worth a comment that we do this explicitly because the env can change and then the release ref would be wrong. Alternatively we could have a getter/setter for the env and if it changes, clear out the ref and the stored class, but that might be overkill.

Reply:
I added a comment to the Class and Object copy operators to make it clear that the ordering is important and that the jobject/jclass should be kept in sync with the JNI enviroment.